tools/fastp/fastp.xml

```diff
@@ -0,0 +1,332 @@
+            <param name="n_base_limit" type="integer" optional="true" label="N base limit" argument="-n" help="If one read's number of N base is >n_base_limit, then this read/pair is discarded. Default is 5."/>
+        </section>
+        <section name="length_filtering_options" title="Length filtering options" expanded="False">
+            <param name="disable_length_filtering" type="boolean" truevalue="-L" falsevalue="" checked="false" label="Disable length filtering" argument="-L" help="Length filtering is enabled by default. If this option is specified, length filtering is disabled."/>
+            <param name="length_required" type="integer" optional="true" label="Length required" argument="-l" help="Reads shorter than this value will be discarded, default is 15."/>
+        </section>
+        <section name="base_correction_options" title="Base correction by overlap analysis options" expanded="False">
+            <param name="correction" type="boolean" truevalue="-c" falsevalue="" checked="false" label="Enable base correction" argument="-c" help="Enable base correction in overlapped regions (only for PE data), default is disabled."/>
+        </section>
+        <section name="umi_processing" title="UMI processing" expanded="False">
+            <param name="umi" type="boolean" truevalue="-U" falsevalue="" checked="false" label="Enable unique molecular identifer" argument="-U" help="Enable unique molecular identifer (UMI) preprocessing."/>
+            <param name="umi_loc" type="text" optional="true" label="UMI location" argument="--umi_loc" help="Specify the location of UMI, can be (index1/index2/read1/read2/per_index/per_read, default is none."/>
+            <param name="umi_len" type="integer" optional="true" label="UMI length" argument="--umi_len" help="If the UMI is in read1/read2, its length should be provided."/>
+            <param name="umi_prefix" type="text" optional="true" label="UMI prefix" argument="--umi_prefix" help="If specified, an underline will be used to connect prefix and UMI (i.e. prefix=UMI, UMI=AATTCG, final=UMI_AATTCG). No prefix by default."/>
+        </section>
+        <section name="overrepresented_sequence_analysis" title="Overrepresented sequence analysis" expanded="False">
+            <param name="overrepresentation_analysis" type="boolean" truevalue="-p" falsevalue="" checked="false" label="Enable overrepresented analysis" argument="-p" help="Enable overrepresented sequence analysis."/>
+            <param name="overrepresentation_sampling" type="integer" optional="true" label="Overrepresentation sampling" argument="-P" help="One in (--overrepresentation_sampling) reads will be computed for overrepresentation analysis (1~10000), smaller is slower, default is 20."/>
+        </section>
+    </inputs>
+
+    <outputs>
+        <data format="fastq" name="out1" label="${tool.name} output"/>
```

Comment:
`${tool.name} output on ${on_string}`  would be useful for seeing which input file was processed

Reply:
done